src/OpenTelemetry.Instrumentation.Dependencies/SqlClientInstrumentationOptions.cs

```diff
@@ -38,5 +49,110 @@ public SqlClientInstrumentationOptions()
         /// Gets or sets a value indicating whether or not the <see cref="SqlClientInstrumentation"/> should capture the text of <see cref="CommandType.Text"/> commands. Default value: False.
         /// </summary>
         public bool CaptureTextCommandContent { get; set; }
+
+        /// <summary>
+        /// Gets or sets a value indicating whether or not the <see cref="SqlClientInstrumentation"/> should parse the DataSource on a SqlConnection into server name, instance name, and/or port connection-level attributes. Default value: False.
+        /// </summary>
+        /// <remarks>
+        /// The default behavior is to set the SqlConnection DataSource as the peer.service tag. If enabled, SqlConnection DataSource will be parsed and the server name will be sent as the net.peer.name or net.peer.ip tag, the instance name will be sent as the db.mssql.instance_name tag, and the port will be sent as the net.peer.port tag if it is not 1433 (the default port).
+        /// </remarks>
+        public bool EnableConnectionLevelAttributes { get; set; } = false;
+
+        internal static SqlConnectionDetails ParseDataSource(string dataSource)
```

Comment:
minor: I'd prefer to move this logic to common helper class and keep options class simple.